packages/cursorless-engine/src/processTargets/targets/BaseTarget.ts

```diff
@@ -194,6 +195,16 @@ export default abstract class BaseTarget<
     return toPositionTarget(this, position);
   }
 
+  toPlainObject(): TargetPlainObject {
+    return {
+      type: this.type,
+      contentRange: rangeToPlainObject(this.contentRange),
+      isReversed: this.isReversed,
+      hasExplicitRange: this.hasExplicitRange,
+    };
+  }
+
+  protected abstract get type(): string;
```

Comment:
OUt of curiosity, why define this one as a getter in the base class rather than just a `readonly` property?

Reply:
I didn't know that you could have an abstract read only property. I have updated now.